Lib/test/test_genericalias.py

```diff
@@ -474,6 +474,40 @@ def test_del_iter(self):
         iter_x = iter(t)
         del iter_x
 
+    def test_paramspec_specialization(self):
+        # gh-124445
+        T = TypeVar("T")
+        U = TypeVar("U")
+        type X[**P] = Callable[P, int]
+
+        generic = X[[T]]
+        self.assertEqual(generic.__args__, ([T],))
+        self.assertEqual(generic.__parameters__, (T,))
+        specialized = generic[str]
+        self.assertEqual(specialized.__args__, ([str],))
+        self.assertEqual(specialized.__parameters__, ())
+
+        generic = X[(T,)]
+        self.assertEqual(generic.__args__, (T,))
```

Comment:
The `__args__` are different here compared to the list version but I think that's correct because `X[(T,)] == X[T]`

Reply:
We need to test a nested version, like this:

```
type X[**P, T] = Callable[P, T]
X[[int, str], float]
X[(int, str), float]
```

Reply:
I, with my limited knowledge, think the `__args__` for a `ParamSpec` should be converted to a tuple here, for the following reasons:

- In the example of PEP 612 the list vs. non-list case is described as equivalent: `hash(Z[[int, str, bool]]) == hash(Z[int, str, bool])` with `class Z(Generic[P])`.
- In general specializing a `ParamSpec` in a `Generic`, `Protocol` or directly in a `Callable` converts the list argument to a tupple. Using a `TypeVar` instead of a `ParamSpec` would not convert the list argument to a tuple in these cases. Conversion to a tuple would mimic this case and be more equivalent.
- Converting it to a tuple in has the benefit that the object will be hashable, as the hash is based on  `__args__`, currently it does not have that benefit.

I've made a separate issue for this https://github.com/python/cpython/issues/124496, not knowing if it could or *should* be addressed at the same time and if it requires further changes beyond this issue.

Reply:
The docs mention that those two forms are "Equivalent", but it's not clear to me whether it's meant as runtime equivalent (in which case I suppose we should flatten the `__args__` in the list case) or type checker equivalent, in which case it's probably ok to keep as-is..

Reply:
You've maybe already seen it. I created a discussion for this: https://discuss.python.org/t/should-list-arguments-of-a-type-statement-that-refer-to-a-paramspec-be-converted-to-a-tuple/66182